### DISCUSSION TITLE
WMS Layers

### DISCUSSION THREAD

**Discussion by sylvain258** [maplibre/maplibre-gl-js#3027] [General]
Hello,

I'd like to add a WMS layer from qgisServer to my web map application. I am looking for the equivalent of that for Mapbox/Maplibre...
I consulted API documentation but I don't understand my layer doesn't work. 

`var wmsLayer = L.tileLayer.wms('myurl.qgs', {
    layers: 'garonne',
    transparent: 'true',
    format: 'image/png',
    maxZoom: 21,
    opacity: 1
}).addTo(map); `

I try this  : 

`  map.addSource('wms-test-source', {
    'type': 'raster',
    url: 'https://linktomyurl.qgs'
});`

`map.addLayer(
    {
        'id': 'wms-test-layer',
        'type': 'raster',
        'source': 'wms-test-source',
        'paint': {}
    }
);`

I don't know if type is correct. It seems I have to declare my layer contained inside, but I don't know where.

Thank you for your answers,

Sylvain

**Comment by HarelM** [comment]
L.something is probably a leaflet example so it's not relevant to this library.
The following is what you are looking for I believe:
https://maplibre.org/maplibre-style-spec/sources/#raster-scheme
Checkout other examples here:
https://maplibre.org/maplibre-gl-js/docs/examples/
